### DISCUSSION TITLE
Consider changing MCU to ESP32?

### DISCUSSION THREAD

**Discussion by rorosaurus** [stevenbennett/DIYson#31] [Ideas]
It would be really cool to consider the ESP32 for this project! I think [Seeed even makes a $5 drop-in replacement you could use!](https://www.seeedstudio.com/Seeed-XIAO-ESP32C3-p-5431.html)

Wifi/BT built-in opens a lot of doors! You could fine tune the color temp or brightness remotely from your phone, integrate the lamp with Google/Siri/Alexa, create schedules, auto-off timers, over-the-air updates, and Home Assistant automation!

As @DimaVIII pointed out, this would solve issue #5 with the LEDC peripheral, since we [have an 80MHz clock we can divide by however many bits of PWM duty resolution we want.](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/peripherals/ledc.html?highlight=pwm#supported-range-of-frequency-and-duty-resolutions)

The ESP32-CAM boards could be something to consider for an all-in-one solution for the color temperature matching feature. It might need a lot of calibration though!

Thanks for considering and for this awesome project (and entertaining videos)!

**Comment by AugieSpear** [comment]
Yea I think this is a good idea, there is an ESP-32 in the same form factor as well.
Here's the [link](https://www.adafruit.com/product/5395)

**Comment by B0rax** [comment]
The ESP32 C3 that you linked does not provide capacitive touch capabilities. The full ESP32 does have those, but is bigger and much more power hungry. (we are talking about 1W idle, which is quite a lot if the light is to be plugged in all the time).

**Reply by rorosaurus** [reply]
> The ESP32 C3 that you linked does not provide capacitive touch capabilities. The full ESP32 does have those, but is bigger and much more power hungry. (we are talking about 1W idle, which is quite a lot if the light is to be plugged in all the time).

Right you are! I can't keep track of all the features Espressif leaves out of all the new chips. -_-

The good news is, there's an original ESP32 with an even smaller footprint! ESP32-PICO-D4 is only 4x4mm, we could totally consider making our own PCB for this project, or using an existing dev board based around the PICO-D4, like the TinyPico (or ideally something cheaper).

[MakerFocus M5Stack M5Stamp Pico](https://www.amazon.com/MakerFocus-M5Stack-M5Stamp-Pico-Development/dp/B0BF5BBP3S/ref=sr_1_10?keywords=PICO-D4&qid=1677380978&s=electronics&sr=1-10): 2pcs for $18

1W idle would be with active wireless communication, IIRC. With deep sleep, or polling wifi updates, we should be able to get a lot lower. Capacitive touch can be used to wake from deep sleep too!

**Reply by AugieSpear** [reply]
You can just use a Sensor IC like [this](https://www.digikey.com/en/products/detail/microchip-technology/AT42QT1012-TSHR/2268867) to get capacitive touch functionality.

**Comment by AugieSpear** [comment]
We could also use a dev board like [this](https://www.adafruit.com/product/5401) to add battery support and have more features. It's a bit big though. Here's an idea, to balance out the weight and size of the housing on each side you could run wires through to the other side of the housing and put some more circuitry like a battery or something.
